utils/s3/credentials_providers/instance_profile_credentials_provider.cc

```diff
@@ -69,7 +69,7 @@ future<> instance_profile_credentials_provider::update_credentials() {
             creds = parse_creds(creds_response);
         },
         http::reply::status_type::ok);
-    ec2_md_logger.info("Retrieved temporary credentials for IAM role: {}", role);
+    co_await http_client.close();
```

Comment:
This lost close is a standalone fix, is it? If so, it deserves a separate patch.

Reply:
done